pkg/cmd/repo/create/create.go

```diff
@@ -855,7 +855,11 @@ func interactiveRepoInfo(client *http.Client, hostname string, prompter iprompte
 		return "", "", "", err
 	}
 
-	visibilityOptions := []string{"Public", "Private", "Internal"}
+	visibilityOptions := []string{"Public", "Private"}
+	// orgs can also create internal repos
+	if owner != "" {
+		visibilityOptions = append(visibilityOptions, "Internal")
+	}
```

Comment:
How might we enhance tests to account for this change?

https://github.com/cli/cli/blob/74b9724c02876b024d03f4c26680ed0ed79860ac/pkg/cmd/repo/create/create_test.go

Reply:
No tests currently exist for the function `interactiveRepoInfo`. I could certainly add tests to cover it, if so desired. I do not think testing this through the addition of tests in `Test_createRun` is prudent.

My preference might be to just ship this then refactor this command to support better coverage of these kinds of changes. What do you think?

Reply:
Talk to me more about why you wouldn't want to test this through `Test_createRun`?

Reply:
Good question 🤔  In addition to this proliferating of a bad testing pattern (as mentioned above), as far as I can tell, the prompter tests are incapable of testing what is being prompted (in this test suite) without [mocking out incorrect user input](https://github.com/cli/cli/blob/4283adbc40d1f0ebad6728feffe30f5a46105140/pkg/cmd/repo/create/create_test.go#L414). Since the user isn't able to select a prompt that doesn't exist, we aren't able to actually detect changes to what is prompted - only testing errors in user input for the prompter. That, to me, smells like over-testing, as the [prompter already covers that use case](https://github.com/AlecAivazis/survey/blob/160123ed7175f77bc8625b59c44b8dc03f5333e3/select_test.go#L308-L328). 

All these taken together means, to me, that the logic we're _actually_ interested in testing is how `visibilityOptions` responds to a change in the return value from the api call. I suppose I could include a test for that, which results in a  tiny refactor. I've committed what that looks like for comparison, here.

Final point: making this change didn't cause any failures to begin with, and given there is the potential for users to unknowingly do something dangerous (set a repo to "Public" when they intended "Private/Internal") with the current behavior I thought that shipping this in favor of a refactor was prudent.

Let me know what you think about the tested commit option I've included 🙂

Reply:
> Outside the scope of this issue, there are some types of organizations that cannot create public repositories like [Enterprise Managed Users](https://docs.github.com/en/enterprise-cloud@latest/admin/managing-iam/understanding-iam-for-enterprises/about-enterprise-managed-users); only noting a similar use case which would require inspection of the selected owner to detect.

A note, @andyfeller, is that this refactor probably makes addressing the above easier as well 🙌

Reply:
My concern about testing the private function with the intention to refactor in the future is that we don't retain any regression safety. However, I think that refactoring this command is such a large piece of work, requiring so much tearing out that the entire test suite will need to be rewritten anyway. I'm content to move forward with your change and to add these tests to the existing table tests because frankly, the existing pattern is extremely annoying.

> Since the user isn't able to select a prompt that doesn't exist, we aren't able to actually detect changes to what is prompted

I do think that we can assert on the `options` inside the prompter test doubled `SelectFunc ` functions e.g.:

```go
p.SelectFunc = func(message, defaultValue string, options []string) (int, error) {
	switch message {
...
	case "Visibility":
		if slices.Contains(options, "Internal") {
			return 0, fmt.Errorf("should not contain internal visibility option: %v", options)
		}
		return prompter.IndexFor(options, "Private")
...
        }
}
```